### ISSUE TITLE
Controller:WaterCoil TemperatureAndHumidityRatio control active when it shouldn't be

### ISSUE THREAD

**Issue by mjwitte** [NREL/EnergyPlus#4994]
User noticed significantly different cooling coil operation with different system sizing options.  The sizing options resulted in nearly a 2x difference in coil size (OnOff vs VT), but the resulting coil operation was significantly different.  Upon further investigation, the culprit is the TemperatureAndHumidityRatio on the Controller:WaterCoil.  In February (using Vancouver BC epw), the humidity ratio entering the cooling coils (there are three vav systems in the defect file) is well below the max humidity ratio setpoint on the cooling coil outlet, and the coil is operating dry, so humidity control should take not action.  But the larger coil is resulting in a lower adjusted leaving setpoint than the smaller coil.  If the control type is changed to Temperature only, then the runs behave exactly the same (even though the coils are sized differently).

**Comment by rraustad** [comment]
The Tsp is always calculated 1) regardless of humrat vs humratmax (sort of) and 2) only once per time step at full water flow rate

It should be calculated each iteration so the approach temp is accurate and calculated at the water flow rate needed to meet the temperature set point. Also, the Tsp should be adjusted only when HRsp < HR. Not sure at this point how the coil solution will converge if this is done every iteration.

```
} else if ( SELECT_CASE_var == iTemperatureAndHumidityRatio ) { // 'TemperatureAndHumidityRatio'
    ControllerProps( ControlNum ).SensedValue = Node( SensedNode ).Temp;
    // Done once per HVAC step
    // WARNING: The scheme for computing the setpoint for the dual temperature and humidity ratio
    //          control strategy breaks down whenever the sensed node temperature is modified by
    //          a controller fired after the current one. Indeed the final sensed node temperature
    //          is likely to have changed in the meantime if the other controller is active,
    //          thereby invalidating the setpoint calculation for the other controller performed
    //          earlier on the air loop.
    if ( ! ControllerProps( ControlNum ).IsSetPointDefinedFlag ) {
        // NOTE: For TEMPANDHUMRAT control the computed value ControllerProps(ControlNum)%SetPointValue
        //       depends on:
        //       - Node(SensedNode)%HumRatMax
        //       - Node(SensedNode)%Temp
        //       - Node(SensedNode)%HumRat
        if ( Node( SensedNode ).HumRatMax > 0 ) {
            // Setpoint can only be computed when the sensed node temperature is evaluated at the max
            // actuated value for the dual humidity ratio / temperature strategy.
            // See routine CalcSimpleController() for the sequence of operations.
            if ( ControllerProps( ControlNum ).NextActuatedValue == RootFinders( ControlNum ).MaxPoint.X ) {
                // Calculate the approach temperature (difference between SA dry-bulb temp and SA dew point temp)
                ApproachTemp = Node( SensedNode ).Temp - PsyTdpFnWPb( Node( SensedNode ).HumRat, OutBaroPress );
                // Calculate the dew point temperature at the SA humidity ratio setpoint
                DesiredDewPoint = PsyTdpFnWPb( Node( SensedNode ).HumRatMax, OutBaroPress );
                // Adjust the calculated dew point temperature by the approach temp
                HumidityControlTempSetPoint = DesiredDewPoint + ApproachTemp;
                // NOTE: The next line introduces a potential discontinuity into the residual function
                //       which could prevent the root finder from finding the root it if were done at each
                //       controller iteration. For this reason we perform the setpoint calculation only
                //       once when the air loop has been evaluated with the max actuated value.
                //       See routine CalcSimpleController() for the sequence of operations.
                ControllerProps( ControlNum ).SetPointValue = min( Node( SensedNode ).TempSetPoint, HumidityControlTempSetPoint ); // Pure temperature setpoint | Temperature setpoint to achieve the humidity ratio setpoint
                // Overwrite the "pure" temperature setpoint with the actual setpoint that takes into
                // account the humidity ratio setpoint.
                // NOTE: Check that this does not create side-effects somewhere else in the code.
                Node( SensedNode ).TempSetPoint = ControllerProps( ControlNum ).SetPointValue;
                // Finally indicate thate the setpoint has been computed
                ControllerProps( ControlNum ).IsSetPointDefinedFlag = true;
            }
        } else {
            // Pure temperature setpoint control strategy
            ControllerProps( ControlNum ).SetPointValue = Node( SensedNode ).TempSetPoint;
            // Finally indicate thate the setpoint has been computed
            ControllerProps( ControlNum ).IsSetPointDefinedFlag = true;
        }
    }
```

**Comment by rraustad** [comment]
Mike hit it exactly. The water coil controller was always setting a lower Tsp. Allowed the controller to always calculate a Tsp while the controller was converging. Might take a few more iterations, but the controller has the correct info to converge now.

Here's a pic of the corrected code. The Temp only simulation resulted in a node SP of about 20C. The TempAndHumRat simulation resulted in a node SP of about 15C. The fixed code results in a SP of about 20C as expected. Note for the defect file, the humidity ratio SP was lowered to test the coil control. The coil increases capacity when the operating humrat exceeds the SP.

![image](https://cloud.githubusercontent.com/assets/5778156/11434735/6d2c39f8-949f-11e5-8fa7-3c35a83022d4.png)

**Comment by rraustad** [comment]
See the excel file for a summary/review in 4994 defect file folder.

**Comment by rraustad** [comment]
PR https://github.com/NREL/EnergyPlus/pull/5336

**Comment by Myoldmopar** [comment]
Closed via #5336
